ComponentKit/TransactionalDataSources/Common/Internal/CKTransactionalComponentDataSourceChangesetModification.mm

```diff
@@ -142,6 +147,26 @@ - (CKTransactionalComponentDataSourceChange *)changeFromState:(CKTransactionalCo
                                                           appliedChanges:appliedChanges];
 }
 
+/**
+ This function computes a pre-changeset-application -> post-changeset-application index path map for updated items.
+ It does a linear pass over all items (in `newState`), and each item is looked up in a previously constructed map (`itemToInitialIndexPathMap`) of item -> pre-changeset-application index path.
+ If it exists we know it was updated and can easily create the mapping (since `newState` is post-changeset-application).
+ */
+static NSDictionary *computeUpdateIndexPaths(NSArray *newState, const std::unordered_map<CKTransactionalComponentDataSourceItem *, NSIndexPath *, CK::HashFunctor<NSObject *>> &itemToInitialIndexPathMap)
+{
+  __block NSMutableDictionary *updatedItemIndexPaths = [NSMutableDictionary dictionary];
+  [newState enumerateObjectsUsingBlock:^(NSArray *items, NSUInteger sectionIdx, BOOL *sectionStop) {
+    [items enumerateObjectsUsingBlock:^(CKTransactionalComponentDataSourceItem *item, NSUInteger index, BOOL *stop) {
```

Comment:
A nested loop like this looks awfully expensive. I thought you said we were already computing this information? Why should core do this? :(